core/modules/wdb/QueryRunner.cc

```diff
@@ -318,8 +318,11 @@ void QueryRunner::_transmit(bool inLast, unsigned int rowCount, size_t tSize) {
         return;
     }
 
+    // Get the czar id number, if there is one.
+    int czarId = (_task->msg->has_czarid()) ? _task->msg->czarid() : -1;
```

Comment:
(Just curious) And what will happen if the identifier will be assigned to `-1`?

Reply:
-1 is perfectly valid as this is only on the worker where czarId is int. -1 was useful in that it indicated the czarId was not from a real source. 0 should work just as well, provided czarIds start at 1.

Reply:
That slipped through. I changed the type to qmeta::CzarId and removed the if statement, since czarId is now required.

If there's a czarId == 0, it doesn't really make much difference, all czar's with czarId == 0 will share the same set of counts/restrictions. This shouldn't happen as long as the database is incrementing the czarId numbers. From this code's point of view, the only thing that matters is that all active czar's have unique id numbers.